react/src/reactTextMask.js

```diff
@@ -19,14 +19,17 @@ export const MaskedInput = React.createClass({
   },
 
   componentDidMount() {
-    const {props, props: {value}} = this
+    this.textMaskInputElement = createTextMaskInputElement()
+    this.update()
+  },
 
-    this.textMaskInputElement = createTextMaskInputElement({inputElement: this.inputElement, ...props})
-    this.textMaskInputElement.update(value)
+  update(value) {
+    const textMaskConfig = Object.assign({inputElement: this.inputElement}, this.props)
```

Comment:
are you able to test the "reactivity"?  It needs a checkbox on the page that can toggle the `guide` (or something).. I tried, but failed.